src/hazelcore/extractions/ocaml/OCamlExtraction_Exp.re

```diff
@@ -0,0 +1,308 @@
+/* Theorem: If extract(ctx, d) = Some(ce, ty) and ctx |- d : ty then ce :_ocaml Typ.extract(ty) */
+
+type t =
+  | OcamlExp(ocaml_exp, HTyp.t)
```

Comment:
Well, actually it's an idea from cyrus, we need it to fully annotate the extracted code (the type will explicitly be declared when a variable is declared). In the implementation, it's necessary because I had to add some patterns into the extraction context, for example, the case rule `| x => x`. It's the simplest way to achieve the full annotation.

Reply:
The very first version of the extraction was a direct translation from the hazel expression to OCaml expression, but it seems cyrus want us to fully-annotate and raise errors on holes. For this reason, I think it's not a good idea to drop `Htyp.t` to make it easier.

Reply:
I will discuss this with Cyrus and get back to you.

Reply:
We need type information sometimes to be able to put in type annotations. So extraction follows the same structure as elaboration from UHExp to DHexp

Reply:
Where does the extraction using the type information?  I can't find it.